app/src/linux-theme-utils.es6

```diff
@@ -0,0 +1,303 @@
+import temp from 'temp';
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+import { execSync } from 'child_process';
+import ini from 'ini';
+
+const Context = {
+  ACTIONS: 'actions',
+  ANIMATIONS: 'animations',
+  APPLICATIONS: 'applications',
+  CATEGORIES: 'categories',
+  DEVICES: 'devices',
+  EMBLEMS: 'emblems',
+  EMOTES: 'emotes',
+  INTERNATIONAL: 'international',
+  MIMETYPES: 'mimetypes',
+  PANEL: 'panel',
+  PLACES: 'places',
+  STATUS: 'status',
+};
+
+const HOME = os.homedir();
+
+const ICON_THEME_PATHS = [
+  path.join(HOME, '.local/share/icons'),
+  path.join(HOME, '.icons'),
+  '/usr/share/icons',
+  '/usr/local/share/icons',
+];
+
+const DESKTOP = process.env.XDG_CURRENT_DESKTOP;
+
+const ICON_EXTENSION = ['.png', '.svg'];
+
+/**
+ * Create a gsettings command string for icons themes
+ *
+ * @returns {string} command
+ * @private
+ */
+function __getGsettingsIconThemeCMD() {
+  let path = __getDesktopSettingsPath();
+  let key = 'icon-theme';
+  return `gsettings get ${path} ${key}`;
+}
+
+/**
+ * Create a gsettings command string for themes
+ *
+ * @returns {string} command
+ * @private
+ */
+function __getGsettingsGtkThemeCMD() {
+  let path = __getDesktopSettingsPath();
+  let key = 'gtk-theme';
+  return `gsettings ${path} ${key}`;
+}
+
+/**
+ * Returns the path for the gsettings command. Currently only GNOME and MATE are supported
+ *
+ * @returns {*}
+ * @private
+ */
+function __getDesktopSettingsPath() {
+  switch (DESKTOP) {
+    case 'MATE':
+      return 'org.mate.interface';
+    case 'GNOME':
+    case 'Budgie:GNOME':
+      return 'org.gnome.desktop.interface';
+    default:
+      return null;
```

Comment:
Hey! It looks like on non gnome / mate environments this will return `null` but `__getGsettingsIconThemeCMD` will still build a command and try to exec `gsettings`. Could we make it so that if this returns null the command is never run? I know we're catching the exec exceptions but it'd be nice if it didn't spawn an invalid command.

Reply:
added checks for null before the command is executed